app/controllers/v0/form210779_controller.rb

```diff
@@ -24,30 +24,39 @@ def create
       end
     rescue JSON::ParserError
       raise Common::Exceptions::ParameterMissing, 'form'
-    rescue => e
-      # Include validation errors when present; helpful in logs/Sentry.
-      Rails.logger.error(
-        'Form210779: error submitting claim',
-        { error: e.message, claim_errors: defined?(claim) && claim&.errors&.full_messages }
-      )
-      raise
     end
 
     def download_pdf
-      claim = SavedClaim::Form210779.find_by!(guid: params[:guid])
-      source_file_path = claim.to_pdf
+      claim = saved_claim_class.find_by!(guid: params[:guid])
+      source_file_path = with_retries('Generate 21-0779 PDF') do
+        claim.to_pdf
+      end
       raise Common::Exceptions::InternalServerError, 'Failed to generate PDF' unless source_file_path
 
       send_data File.read(source_file_path),
-                filename: "21-0779_#{SecureRandom.uuid}.pdf",
+                filename: download_file_name(claim),
                 type: 'application/pdf',
                 disposition: 'attachment'
+    rescue ActiveRecord::RecordNotFound
+      raise Common::Exceptions::RecordNotFound, params[:guid]
     ensure
-      File.delete(source_file_path) if source_file_path && File.exist?(source_file_path)
+      File.delete(source_file_path) if defined?(source_file_path) && source_file_path && File.exist?(source_file_path)
     end
 
     private
 
+    def filtered_params
+      params.require(:form)
+    end
+
+    def download_file_name(claim)
+      "21-0779_#{claim.veteran_name.gsub(' ', '_')}.pdf"
+    end
```

Comment:
Be nice to put this in a helper or presenter, or somewhere it could be reusable so we don't have to copy/paste the method.